proto/dydxprotocol/vault/tx.proto

```diff
@@ -39,6 +40,42 @@ message MsgDepositToVault {
 // MsgDepositToVaultResponse is the Msg/DepositToVault response type.
 message MsgDepositToVaultResponse {}
 
+// MsgWithdrawFromVault attempts to withdraw the specified target amount of
+// asset from the vault to the subaccount.
+message MsgWithdrawFromVault {
+  option (cosmos.msg.v1.signer) = "subaccount_id";
+
+  // The vault to withdraw from.
+  VaultId vault_id = 1;
+
+  // The subaccount to withdraw to.
+  dydxprotocol.subaccounts.SubaccountId subaccount_id = 2;
+
+  // The target amount of quote quantums to withdraw.
+  // The target amount should account for slippage. For example:
+  //    * user equity = $1020
+  //    * target amount = $1000
+  //    * slippage = $50
+  // then, the amount withdrawn is $970 (= 1020 - (1000 + 50 - 1020))
+  bytes target_quote_quantums = 3 [
+    (gogoproto.customtype) =
+        "github.com/dydxprotocol/v4-chain/protocol/dtypes.SerializableInt",
+    (gogoproto.nullable) = false
+  ];
+}
+
+// MsgWithdrawFromVaultResponse is the Msg/WithdrawFromVault response type.
+message MsgWithdrawFromVaultResponse {
+  // Number of shares that have been redeemed as part of the withdrawal.
+  NumShares redeemed_shares = 1;
+
+  // Number of shares remaining after the withdrawal.
+  NumShares remaining_shares = 2;
+
+  // Number of shares for the vault after the withdrawal.
+  NumShares total_shares = 3;
```

Comment:
I assume one is the remaining shares of the individual, and one is the remaining shares in the whole vault? If so, I think this distinction could be clearer

Reply:
+1. We can probably not include `total_shares` in the response?

Reply:
To clarify:
1. `redeemed_shares`: given that user specifies USDC, this indicates the # of shares that were redeemed for the user
2. `remaining_shares`: user's shares after the withdrawal
3. `total_shares`: shares for the entire vault

I think we need all three for the FE to calculate the new vault equity + user equity in the vault.

Technically, FE will have some of these values cached, but things could have changed in between so i think it's better to return all these so FE can calculate the amount using these values

Reply:
From the protocol level, does it seem a bit unexpected UX to input `quote_quantums` and in response, get `shares`?

Reply:
we are using shares now per offline discussion